lib/decidim/participatory_documents/component.rb

```diff
@@ -28,6 +28,21 @@
   component.settings(:step) do |settings|
   end
 
+  component.exports :suggestions do |exports|
+    exports.collection do |component_instance|
+      suggestion_ids = Decidim::ParticipatoryDocuments::Suggestion.joins(
+        "INNER JOIN decidim_participatory_documents_documents ON decidim_participatory_documents_documents.id = decidim_participatory_documents_suggestions.suggestable_id " \
+        "INNER JOIN decidim_components ON decidim_components.id = decidim_participatory_documents_documents.decidim_component_id"
+      )
+      Decidim::ParticipatoryDocuments::Suggestion
+        .where(id: suggestion_ids).uniq
+        .filter { |suggestion| suggestion.component == component_instance }
```

Comment:
this export has two problems:
- It my return incorrect suggestions as it is checking  `decidim_participatory_documents_documents.id` against the `suggestable_id` but it does not take into account the title.
- It does not include the suggestions from sections
It also is not very efficient, we are querying all the results for all the components and then filtering in the application level (this forces to download unnecessary data from the database), it is better to filter it as a `where` query instead of using a filter.

Her is a suggestion:

```ruby
#NOTE: if there's ever more than one document per component this needs to be updated
document = Decidim::ParticipatoryDocuments::Document.find_by(component: component_instance)
Decidim::ParticipatoryDocuments::Suggestion.where(suggestable: document).or(Decidim::Parti
cipatoryDocuments::Suggestion.where(suggestable: document.sections))
```